### PR TITLE
`anyhow` Error

### PR DESCRIPTION
**Pull request by burrbull**
https://github.com/rust-embedded/svd/pull/98

### CODE REVIEW

**Comment by burrbull** [comment]
cc @therealprof 

Looks like it works:
```
+ /home/burrbull/builds/svd2rust/release/svd2rust --target riscv -i k210.svd
ERROR svd2rust: In peripheral `FFT`
ERROR svd2rust: caused by: In register `ctrl`
ERROR svd2rust: caused by: Parsing field #0
ERROR svd2rust: caused by: In field `point`
ERROR svd2rust: caused by: Expected content in <enumeratedValues> tag, found none
ERROR svd2rust: backtrace: disabled backtrace
```

**Comment by therealprof** [comment]
Sweet!

File: src/lib.rs [hunk 1]
```diff
@@ -461,18 +458,20 @@ pub struct DeviceSpecific {
...
 }
 
-type Result<T> = std::result::Result<T, SvdError>;
-#[derive(Debug)]
+use anyhow::Result;
+#[derive(Debug, thiserror::Error)]
 pub enum SvdError {
+    #[error("Cannot format crate")]
     Fmt,
+    #[error("Cannot render SVD device")]
```
Comment (Emilgardis) [inline src/lib.rs]: _cannot_ vs _couldn't_ vs _can not_

I prefer _cannot_ for this

File: Cargo.toml [hunk 2]
```diff
@@ -35,13 +35,16 @@ path = "src/main.rs"
...
 inflections = "1.1.0"
 log = { version = "~0.4", features = ["std"] }
 quote = "1.0"
 proc-macro2 = "1.0"
+anyhow = "1.0.19"
+thiserror = "1.0.5"
 
 [dependencies.svd-parser]
+git = "https://github.com/rust-embedded/svd"
+branch = "anyhow"
```
Comment (Emilgardis) [inline Cargo.toml]: Assuming this is just for the draft.

We could use the new git+version support added for publishing crates added recently, but I'm not sure myself how to properly use it.

File: src/main.rs [hunk 3]
```diff
@@ -79,20 +76,20 @@ fn run() -> Result<()> {
     match matches.value_of("input") {
         Some(file) => {
             File::open(file)
-                .chain_err(|| "couldn't open the SVD file")?
+                .context("couldn't open the SVD file")?
                 .read_to_string(xml)
-                .chain_err(|| "couldn't read the SVD file")?;
+                .context("couldn't read the SVD file")?;
```
Comment (Emilgardis) [inline src/main.rs]: Missed this when implementing, should probably be uppercase "C"

File: src/main.rs [hunk 4]
```diff
@@ -79,20 +76,20 @@ fn run() -> Result<()> {
...
         }
         None => {
             let stdin = std::io::stdin();
             stdin
                 .lock()
                 .read_to_string(xml)
-                .chain_err(|| "couldn't read from stdin")?;
+                .context("couldn't read from stdin")?;
```
Comment (Emilgardis) [inline src/main.rs]: See above

File: src/generate/peripheral.rs [hunk 5]
```diff
@@ -600,7 +600,7 @@ fn expand_cluster(cluster: &Cluster, defs: &RegisterProperties) -> Result<Vec<Re
     let defs = cluster.default_register_properties.derive_from(defs);
 
     let cluster_size = cluster_size_in_bits(cluster, &defs)
-        .chain_err(|| format!("Cluster {} has no determinable `size` field", cluster.name))?;
+        .context(format!("Cluster {} has no determinable `size` field", cluster.name))?;
```
Comment (Emilgardis) [inline src/generate/peripheral.rs]: Use `with_context`

Comment (Emilgardis) [reply src/generate/peripheral.rs]: Think there are some other places in rust-embedded/svd#98 that could do the same, but I'm sure it doesnt really matter in optimized builds.

File: src/main.rs (hunk 3 above)
Comment (Emilgardis) [reply src/main.rs]: _cannot_ vs _couldn't_ vs _can not_

I prefer _cannot_ for this

**Comment by Emilgardis** [comment]
We can merge this as if, if we do we'd bump the required cargo version to publish, but not to compile (published) crate.

**Comment by burrbull** [comment]
First we need make new release of `svd-parser` to remove git dependency.

**Comment by burrbull** [comment]
By the way what MSRV version now? Can we merge #381 ?

**Comment by therealprof** [comment]
@burrbull It changes the MSRV of the PACs to 1.37? I don't think we want to do that just yet. Bumping the MSRV of svd or svd2rust would be fine.

**Comment by burrbull** [comment]
bors try

**Comment by burrbull** [comment]
bors try

**Comment by burrbull** [comment]
`std::backtrace::Backtrace` still unstable.

**Comment by burrbull** [comment]
bors try

**Comment by burrbull** [comment]
@therealprof 

Now it works on stable, but without backtrace.

If compile `svd2rust` on nightly, backtrace works too.

**Comment by therealprof** [comment]
@burrbull Can you rebase this, fix the conflicts and add a ChangeLog entry?

**Comment by burrbull** [comment]
Looks like [std::backtrace::Backtrace](https://doc.rust-lang.org/std/backtrace/struct.Backtrace.html) not stable yet.

**Comment by burrbull** [comment]
Rebased

**Comment by burrbull** [comment]
bors try

**Pull request merged**